specification/walkthrough.md

```diff
@@ -0,0 +1,263 @@
+
+*BIDS Stats Models* proposes a general machine-readable document to describe multi-stage neuroimaging analyses in a precise, yet flexible manner. We accomplish this by defining a *graph* composed of **Nodes** representing each level of the analysis, and `Edges` which define the flow of data from one `Node` to another. Within each {py:attr}`~bsmschema.models.Node` we specify a {py:attr}`~bsmschema.models.Model` to estimate, and at least one {py:attr}`~bsmschema.models.Contrast` to define the computed outputs of each `Node`. Within each node we also specify how to group the incoming inputs into analysis units using the {py:attr}`~bsmschema.models.Node.GroupBy` directive.
+
+
+## A simple example
+
+In a [Simon task](https://openneuro.org/datasets/ds000101/versions/00004), participants were scanned for 2 runs and asked to indicate whether a diamond that was presented to the left or right of a central fixation cross was green or red. There were two conditions: color-spatial *congruent* and *incongruent* trials.
+
+A basic analysis of this task is to determine which regions showed *greater activity for incongruent versus congruent trials*, across all participants. 
+
+We can perform this analysis by first estimating a **run-level** timeseries model for "Incongruent" and "Congruent" trials--separately for each individual run. We then compute a contrast comparing *Incongruent > Congruent* (IvC) trials. Next, we pass the resulting statistical maps for the contrast to a **subject-level** estimator, which computes the average *IvC* effect for each subject separately. Finally, we pass the resulting estimates to a **dataset-level** estimator, which conduts a one-sample t-test across all of the subject estimates for the *IvC* contrast.
+
+Let's visualize this model for 3 participants:
+
+![flow](images/flow_model.png)
+
+We can formally represent this analysis as **BIDS Stats Model**:
+
+```json
+{
+  "Name": "Simon IvC",
+  "BIDSModelVersion": "1.0.0",
+  "Input": {"Subject": ["01", "02", "03"]},
+  "Nodes": [
+    {
+      "Level": "Run",
+      "Name": "run_level",
+      "GroupBy": ["run", "subject"],
+      "Model": {"X": [1, "incongruent", "congruent"], "Type": "glm"},
+      "Contrasts": [
+        {
+          "Name": "IvC",
+          "ConditionList": ["incongruent", "congruent"],
+          "Weights": [1, -1],
+          "Test": "t"
+        }
+      ]
+    },
+    {
+      "Level": "Subject",
+      "Name": "subject_level",
+      "GroupBy": ["subject", "contrast"],
+      "Model": {"X": [1], "Type": "Meta"},
```

Comment:
```suggestion
      "Model": {"X": [1], "Type": "meta"},
```

I would suggest storing this model in examples, where it can be easily validated, and then include it via a directive. See the example in guide.md

Reply:
I happened to do that just before you reviewed I think